control-plane-operator/controllers/hostedcontrolplane/ingress/router.go

```diff
@@ -247,25 +247,14 @@ func ReconcileRouterService(svc *corev1.Service, kasPort int32, internal, crossZ
 	svc.Spec.Type = corev1.ServiceTypeLoadBalancer
 	svc.Spec.Selector = hcpRouterLabels()
 	foundHTTPS := false
-	foundKAS := false
 
-	// TODO (alberto): why this criteria?
-	// Introduced here https://github.com/openshift/hypershift/pull/1614/files#diff-62c16653415b8d89921cb26796abc479c31da1654095f7c46b551b470533d66dR368-R372.
-	if kasPort == 443 {
-		foundKAS = true
-	}
```

Comment:
I'm not familiar with the use-case, but is this so that someone could separate router/KAS traffic onto different ports? If so, this would just be a guard for the ROSA HCP use-case where we use the same port for both. We would then need the `if !foundKas` block further down at line 279

Reply:
>I'm not familiar with the use-case, but is this so that someone could separate router/KAS traffic onto different ports? If so, this would just be a guard for the ROSA HCP use-case where we use the same port for both. We would then need the if !foundKas block further down at line 279

Might be the original reason, I'm not aware of this use atm.
But also the `if kasPort == 443 {foundKAS = true}` is not a valid criteria any more for that potential use case.